### PR TITLE
feat: add `clientIdentifier` option to `createServer`

### PR DESCRIPTION
**Pull request by JKRhb**
This PR is a follow-up to #190. It fixes a couple of issues that were present on that branch, cleans up the commit history, and removes the added `package.json` file.

### CODE REVIEW

File: test/server.js [hunk 1]
```diff
@@ -1263,4 +1263,104 @@ describe('server block cache', function() {
     })
   })
 
-})
\ No newline at end of file
+})
+
+describe('server LRU', function() {
```
Comment (JKRhb) [inline test/server.js:1268]: I am not sure about the naming for these test cases. In [line 1152](https://github.com/mcollina/node-coap/pull/256/files#diff-d14cbfb50ed1cad7db5f4fef6a6076961b7cc9be980a3be06a70998f0eb8ebceR1152), there is already a `server LRU` present. Maybe the tests can be merged?

**Comment by JKRhb** [comment]
@Apollon77 Could you maybe have a quick look on this one? :) Maybe it could be integrated into the 0.25.0 release.

**Comment by Apollon77** [comment]
Sure, I ran a bit outt of sync with some tasks ... will check tonight

**Comment by JKRhb** [comment]
> will check tonight

Great, thank you!

File: test/server.js (hunk 1 above)
Comment (Apollon77) [reply test/server.js:1268]: Merging is one option, having the name state an "whats the test focus on" infgo is also an idea. tests should not become too complex imo

**Comment by Apollon77** [comment]
@JKRhb Basically ok, I left you one comment for the test question ... what you think?

File: test/server.js (hunk 1 above)
Comment (JKRhb) [reply test/server.js:1268]: > , having the name state an "whats the test focus on" infgo is also an idea. tests should not become too complex imo

Yeah, you are right. What do you think about this one?

```suggestion
describe('Client Identifier', function() {
```

 (Not sure why I haven't been thinking about this name earlier 😅)

**Comment by JKRhb** [comment]
> @JKRhb Basically ok, I left you one comment for the test question ... what you think?

Thank you :) I replied with a naming suggestion :)

**Comment by Apollon77** [comment]
accepted ... lets formally have test running and then have fun merging it :-)

**Comment by JKRhb** [comment]
> accepted ... lets formally have test running and then have fun merging it :-)

Awesome :) Thanks for the review!

**Comment by Apollon77** [comment]
> Thanks for the review!

yw

**Pull request merged**